File: okj/rendszeruzemelteto/eucsatlakozas/java/EU.java
import java.io.*;
import java.nio.charset.*;
import java.nio.file.*;
import java.time.*;
import java.util.*;

public class EU {

    public static void main(String[] args) throws IOException {
        var csatlakozasok = new ArrayList<Csatlakozas>();
        var lines = Files.readAllLines(Path.of("EUcsatlakozas.txt"), StandardCharsets.ISO_8859_1);

        for(var line : lines) {
            csatlakozasok.add(new Csatlakozas(line));
        }

        System.out.println("3. Feladat: 2018-ig csatlakozott országok száma: " + csatlakozasok.size());

        var tagallamok2007ben = 0;
        for(var csati : csatlakozasok) {
            if(csati.datum.getYear() == 2007) {
                ++tagallamok2007ben;
            }
        }

        System.out.println("4. Feladat: 2007-ben csatlakozott országok száma: " + tagallamok2007ben);

        for(var csati : csatlakozasok) {
            if(csati.orszag.equals("Magyarország")) {
                System.out.println("5. Feladat: Magyarország csatlakozása: " + csati.datum);
                break;
            }
        }

        var voltEMajusban = false;
        for(var csati : csatlakozasok) {
            if(csati.datum.getMonth() == Month.MAY) {
                voltEMajusban = true;
                break;
            }
        }

        System.out.println("6. Feladat: " + (voltEMajusban ? "Volt" : "Nem volt") + " májusban csatlakozás");

        var utolso = csatlakozasok.get(0);
        for(var csati : csatlakozasok) {
            if(csati.datum.isAfter(utolso.datum)) {
                utolso = csati;
            }
        }

        System.out.println("7. Feladat: Utoljára csatlakozott: " + utolso.orszag);
        System.out.println("8. Feladat:");

        var stat = new HashMap<Integer, Integer>();
        for(var csati : csatlakozasok) {
            var ev = csati.datum.getYear();

            stat.put(ev, stat.getOrDefault(ev, 0) + 1);
        }

        for(var entry : stat.entrySet()) {
            System.out.println(entry.getKey() + " - " + entry.getValue() + " db ország");
        }
    }
}

File: okj/rendszeruzemelteto/eucsatlakozas/c#/EU.cs
using System;
using System.Collections.Generic;
using System.IO;

var csatlakozasok = new List<Csatlakozas>();
var lines = File.ReadAllLines("EUcsatlakozas.txt", System.Text.Encoding.GetEncoding("ISO-8859-1"));

foreach(var line in lines) {
    csatlakozasok.Add(new Csatlakozas(line));
}

Console.WriteLine($"3. Feladat: 2018-ig csatlakozott országok száma: {csatlakozasok.Count}");

var tagallamok2007ben = 0;
foreach(var csati in csatlakozasok) {
    if(csati.datum.Year == 2007) {
        ++tagallamok2007ben;
    }
}

Console.WriteLine($"4. Feladat: 2007-ben csatlakozott országok száma: {tagallamok2007ben}");

foreach(var csati in csatlakozasok) {
    if(csati.orszag == "Magyarország") {
        Console.WriteLine($"5. Feladat: Magyarország csatlakozása: {csati.datum}");
        break;
    }
}

var voltEMajusban = false;
foreach(var csati in csatlakozasok) {
    if(csati.datum.Month == 5) {
        voltEMajusban = true;
        break;
    }
}

Console.WriteLine("6. Feladat: " + (voltEMajusban ? "Volt" : "Nem volt") + " májusban csatlakozás");

var utoljaraCsatlakozo = csatlakozasok[0];
foreach(var csati in csatlakozasok) {
    if(csati.datum > utoljaraCsatlakozo.datum) {
        utoljaraCsatlakozo = csati;
    }
}

Console.WriteLine($"7. Feladat: Utoljára csatlakozott: {utoljaraCsatlakozo.orszag}");
Console.WriteLine("8. Feladat:");

var stat = new Dictionary<int, int>();
foreach(var csati in csatlakozasok) {
    var ev = csati.datum.Year;

    stat[ev] = stat.GetValueOrDefault(ev, 0) + 1;
}

foreach(var (ev, dbSzam) in stat) {
    Console.WriteLine($"{ev} - {dbSzam} db ország");
}